src/tests/utils/Load.test.js

```diff
@@ -29,6 +30,20 @@ describe('#unzipDependencies', () => {
     expect(getSpy).toBeCalled();
     expect(getSpy).toBeCalledWith('https://packages.fhir.org/hl7.fhir.r4.core/4.0.1', callbackFunction);
   });
+
+  //TODO - test failed http request
+
+  // it('should add failed http requests to a list of emptyDependencies', async () => {
+  //   getSpyBad = jest.spyOn(http, 'get').mockImplementation(() => {
+  //     const res = { statusCode: 404 };
+  //     return res;
+  //   });
+  //   const unzipReturn = await unzipDependencies(resources, 'hello', '123');
+  //   const callbackFunction = getSpy.mock.calls[0][1];
+  //   expect(getSpyBad).toBeCalled();
+  //   expect(getSpyBad).toBeCalledWith('https://packages.fhir.org/hello/123', callbackFunction);
+  //   expect(unzipReturn).toBe({ resourceArr: resources, emptyDependencies: ['hello#123'] });
+  // });
```

Comment:
I think I've figured out a valid way to test this. If someone else knows a better way, definitely still open to hearing it, but this is what I've got:
```javascript
it('should add failed http requests to a list of emptyDependencies', () => {
    const unzipPromise = unzipDependencies(resources, 'hello', '123');
    // Get the callback, and run it with a 404
    const callbackFunction = getSpy.mock.calls[0][1];
    callbackFunction({ statusCode: 404 });
    expect(getSpy).toBeCalled();
    expect(unzipPromise).resolves.toEqual({ emptyDependencies: ['hello123'], resourceArr: [] });
  });
```
Note that this code will require you remove the `tarSpy`, but as far as I could tell the `tarSpy` was not necessary anyway. What this code is doing is mocking the entire `http.get`, and then since we can get the arguments that a function is called with, we get the second argument, which is actually a callback function. Then we manually trigger that callback with a 404 status code, and we can test that the `emptyDependencies` array is indeed populated.